components/datetime/src/pattern/mod.rs

```diff
@@ -16,6 +16,28 @@ pub enum PatternItem {
     Literal(String),
 }
 
+/// The granularity of time represented by `PatternItem`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum TimeGranularity {
+    Hours,
+    Minutes,
+    Seconds,
+}
+
+impl PatternItem {
+    fn time_granularity(&self) -> Option<TimeGranularity> {
+        match self {
+            Self::Field(field) => match field.symbol {
+                fields::FieldSymbol::Hour(_) => Some(TimeGranularity::Hours),
+                fields::FieldSymbol::Minute => Some(TimeGranularity::Minutes),
+                fields::FieldSymbol::Second(_) => Some(TimeGranularity::Seconds),
+                _ => None,
+            },
+            _ => None,
+        }
+    }
+}
```

Comment:
I don't think having it here has any performance/size implications, but I think from the architectural standpoint that method located here is misplaced.
It communicates to me that there's a need to calculate `time_granularity` for `PatternItem` elements, while in fact, there's only one scenario (so far?) where we need it and it's an outlier.

For cases where a bunch of code is used only for a specialized scenario, it is a good idea to:
* Keep that code isolated in some helper function/module
* Link it only from the direct code that needs it. In this case, call it only from `b` match case. This will allow people later to easily grep on why the function exists, where it is used and either find a callee or a caller when needed.
* Document the shit out of that special function (`get_time_granularity_for_pattern`) so that people understand what it does and why when they reach that code

Reply:
Thank you for your feedback (to this and related comments). I'm totally happy to, and was planning to reorganize a lot of this. As I said in https://github.com/unicode-org/icu4x/pull/444#issuecomment-765809418, I just wanted to do a first pass from a path of least resistance to ensure that the new functionality was tested (I love having tests to back me up).

Your comments give me good direction for how to rearrange all of this.